completion_aggregator/api/v1/views.py

```diff
@@ -384,22 +385,23 @@ def get(self, request, course_key):
         root_block = request.query_params.get('root_block')
         if root_block:
             root_block = UsageKey.from_string(root_block).map_into_course(course_key)
-        if is_stale:
+
+        if is_stale and waffle.flag_is_active(request, 'completion_aggregator.aggregate_stale_from_scratch'):
             aggregator_queryset = []
         else:
             aggregator_queryset = self.get_queryset().filter(
                 course_key=course_key,
                 user__in=[enrollment.user for enrollment in paginated],
-            )
+            ).select_related('user')
```

Comment:
This, of course, is the big time saver here.  It was not just making an extra query per user, but an extra query per individual aggregator, which amounts to several dozen extra queries for a single user in a moderate-sized course.